collectors/ebpf.plugin/ebpf.c

```diff
@@ -1092,12 +563,147 @@ static void parse_args(int argc, char **argv)
     }
 }
 
+/**
+ * Fill the ebpf_functions structure with default values
+ *
+ * @param ef the pointer to set default values
+ */
+void fill_ebpf_functions(ebpf_functions_t *ef) {
+    memset(ef, 0, sizeof(ebpf_functions_t));
+    ef->kernel_string = kernel_string;
+    ef->running_on_kernel = running_on_kernel;
+    ef->map_fd = callocz(EBPF_MAX_MAPS, sizeof(int));
+    ef->isrh = isrh;
+}
+
+/**
+ * Define how to load the ebpf programs
+ *
+ * @param ptr the option given by users
+ */
+static inline void how_to_load(char *ptr) {
+    if (!strcasecmp(ptr, "return"))
+        ebpf_set_thread_mode(MODE_RETURN);
+}
+
+/**
+ * Set collector values
+ */
+static void set_collector_values() {
+    struct section *sec = collector_config.first_section;
+    int disable_apps = 0;
+    int enabled = 0;
+    while(sec) {
+        struct config_option *values;
+        if(!strcasecmp(sec->name, "global")) {
+            values = sec->values;
+            while(values) {
+                if (!strcasecmp(values->name, "load"))
+                    how_to_load(values->value);
+                else if(!strcasecmp(values->name, "disable apps")) {
+                    if (!strcasecmp(values->value, "yes")) {
+                        ebpf_disable_apps();
+                        disable_apps = 1;
+                    }
+                }
+
+                values = values->next;
+            }
+        } else if(!strcasecmp(sec->name, "global")) {
+            values = sec->values;
+            while(values) {
+                if (!strcasecmp(values->name, "process")) {
```

Comment:
I would prefer us here to go trough the list of availible modules as oposed to hardcoded? Maybe add the `.config_name` into `ebpf_modules` and go trough that aray here?